### ISSUE TITLE
[Feature] Model Context Protocol

### ISSUE THREAD

**Issue by crazyyanchao** [lobehub/lobe-chat#4945]
### 🥰 Feature Description

 AS MCP Host replacing Claude desktop.

### 🧐 Proposed Solution

Check the community server progress @ https://github.com/punkpeye/awesome-mcp-servers

**Comment by arvinxx** [comment]
we are planning to make a desktop app in the next year. Maybe Jan or Feb. MCP will be a core feature. Stay tuned!

**Comment by punkpeye** [comment]
Once you guys are ready, let me know, and we will list it on https://glama.ai/mcp/clients

**Comment by CXwudi** [comment]
Would love to see MCP support on self-hosted LobeChat

**Comment by cl1107** [comment]
> we are planning to make a desktop app in the next year. Maybe Jan or Feb. MCP will be a core feature. Stay tuned!

Has it been started yet?

**Comment by arvinxx** [comment]
@cl1107 when we start the desktop version, it starts 🙂

**Comment by 18651619390** [comment]
> we are planning to make a desktop app in the next year. Maybe Jan or Feb. MCP will be a core feature. Stay tuned!

is there include ios app?

**Comment by zhufeizzz** [comment]
@arvinxx 同催一下MCP，也不一定要绑定桌面版吧，SSE接口可以先用着

**Comment by mjohenneken** [comment]
yeah agreed, setup with sse should be possible without a desktop app e.g. when installed as PWA. Has someone worked on using the Plugin as a wrapper of mcp sse server?

**Comment by 18651619390** [comment]
我也希望不开发桌面应用，我们都把Lobe-chat当作一个agent来使用了，更多的桌面程序会影响体验感

**Comment by tianautonerd** [comment]
I think you guys could integrate the MCP with Lobe's plugin feature. In this way, it will unleash the superpower of Lobe and inspire more innovations in Lobe's open-source community!

**Comment by CXwudi** [comment]
> I think you guys could integrate the MCP with Lobe's plugin feature. In this way, it will unleash the superpower of Lobe and inspire more innovations in Lobe's open-source community!

The Lobe plugin requires you to have a backend and public URL for it. I don't want to share my domain name towards the public and maintain it myself.

**Comment by arvinxx** [comment]
SSE interface will be replaced by the new Streamable HTTP, so it's not a good idea for us to intergate it now. Our desktop application will be released in about two weeks, stay tuned!

**Comment by joe200631** [comment]
> SSE interface will be replaced by the new Streamable HTTP, so it's not a good idea for us to intergate it now. Our desktop application will be released in about two weeks, stay tuned!

Is the desktop application release support MCP?

**Comment by arvinxx** [comment]
> Is the desktop application release support MCP?

Yes

**Comment by AlgorithmicKing** [comment]
so mcp is now in lobechat? or still in dev?

**Comment by leonwangcn** [comment]
真的很需要这个能力

**Comment by yincangshiwei** [comment]
@arvinxx 虽然期待桌面应用程序版的效果，但也希望考虑下web端里能够使用MCP，因为web端对比桌面端来说会更便捷一些。喜欢lobechat就是因为它提供的web端使用。

**Comment by CXwudi** [comment]
+1 同感，如果我非得通过桌面端才能用上MCP那为什么我不去用Cherry Studio

**Comment by gkzhb** [comment]
Open WebUI 出了一个 https://github.com/open-webui/mcpo 作为 mcp 网关接入，感觉可以借鉴一下

**Comment by 18651619390** [comment]
同样非常不希望再多一个客户端，这很不方便跟我们使用lobechat的初衷相违背

**Comment by dididiandao** [comment]
我倒是希望出客户端

**Comment by gpt4thewin** [comment]
I think Lobechat urgently needs to implement MCP. This is being adopted everywhere really fast

**Comment by arvinxx** [comment]
> @arvinxx 虽然期待桌面应用程序版的效果，但也希望考虑下web端里能够使用MCP，因为web端对比桌面端来说会更便捷一些。喜欢lobechat就是因为它提供的web端使用。

Web 端会考虑在 streamable HTTP 推出后适配。sse很快就会被废弃的

**Comment by DBFritz** [comment]
I created a plugin for using MCP servers with lobechat, you can find the repo in [DBFritz/lobechat-mcp-plugin](https://github.com/DBFritz/lobechat-mcp-plugin).

You should be able to run it with
```
npx -y lobechat-mcp-plugin
```
but previously you need to define a `config.json` file.

From the lobechat interface you can add the custom plugin by providing the manifest `http://localhost:3000/<identifier>/manifest.json`

It supports stdio, sse and you can even define an OpenAPI url.  It was for a personal project but I think someone else can find it useful.

**Comment by DBFritz** [comment]
I created a plugin for using MCP servers with lobechat, you can find the repo in [DBFritz/lobechat-mcp-plugin](https://github.com/DBFritz/lobechat-mcp-plugin).

You should be able to run it with
```
npx -y lobechat-mcp-plugin
```
but previously you need to define a `config.json` file.

From the lobechat interface you can add the custom plugin by providing the manifest `http://localhost:3000/<identifier>/manifest.json`

It supports stdio, sse and you can even define an OpenAPI url.  It was for a personal project but I think someone else can find it useful.

**Comment by johnlanni** [comment]
> Open WebUI 出了一个 https://github.com/open-webui/mcpo 作为 mcp 网关接入，感觉可以借鉴一下

我感觉这个路子有点歪，应该通过 remote mcp server 来解决浏览器接入的问题，local mcp server 也可以通过 stdio 转 HTTP 来实现。

**Comment by CXwudi** [comment]
我感觉楼上@DBFritz的MCP转LobeChat插件的工具有搞头

**Comment by gpt4thewin** [comment]
To the Lobechat contributors community, please you need to implement MCP asap !!

**Comment by AlgorithmicKing** [comment]
> we are planning to make a desktop app in the next year. Maybe Jan or Feb. MCP will be a core feature. Stay tuned!

so mcp will come next year? why cant you add mcp as experimental feature first and then plan the big release next year?

**Comment by arvinxx** [comment]
> > we are planning to make a desktop app in the next year. Maybe Jan or Feb. MCP will be a core feature. Stay tuned!
> 
> so mcp will come next year? why cant you add mcp as experimental feature first and then plan the big release next year?

This year

**Comment by arvinxx** [comment]
v1.82.0 have supported the Streamable HTTP mcp server.
